### PR TITLE
将rt_mutex_init()由RT_IPC_FLAG_FIFO改为RT_IPC_FLAG_PRIO

### PR DESCRIPTION
**Pull request by mysterywolf**
Signed-off-by: Meco Man <920369182@qq.com>

## 拉取/合并请求描述：(PR description)

[
边缘化RT_IPC_FLAG_FIFO
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

**Comment by mysterywolf** [comment]
请无视format错误

**Review by Guozhanxin — APPROVED** [review]

File: components/libc/compilers/common/unistd.c [hunk 1]
```diff
@@ -41,7 +41,7 @@ char *ttyname(int fd)
 }
 RTM_EXPORT(ttyname);
 
-unsigned int sleep(unsigned int seconds)
+RT_WEAK unsigned int sleep(unsigned int seconds)
```
Comment (Guozhanxin) [inline components/libc/compilers/common/unistd.c:44]: 这些地方还是要加 RT_WEAK 吗？

File: components/libc/compilers/common/unistd.c [hunk 2]
```diff
@@ -53,7 +53,7 @@ unsigned int sleep(unsigned int seconds)
 }
 RTM_EXPORT(sleep);
 
-int usleep(useconds_t usec)
+RT_WEAK int usleep(useconds_t usec)
```
Comment (Guozhanxin) [inline components/libc/compilers/common/unistd.c:56]: 还有这里？

Comment (mysterywolf) [reply components/libc/compilers/common/unistd.c:56]: 这是个意外 这个是其他pr的残余 不小心提交上来了

**Comment by BernardXiong** [comment]
Please fix this issue: 'Check File Format and License / Scan code format and license (pull_request) '.

**Comment by mysterywolf** [comment]
修复不了，这个不是我的问题引起的，我只是改了一个flag，这个是被改文件其他位置引起的。改CI会扫描整个文件，只要文件一处不符合规定就会报错。CI只是提供一个参考，如果修改无误就可以人工合并了。

**Comment by mysterywolf** [comment]
https://github.com/RT-Thread/rt-thread/issues/4704

**Comment by BernardXiong** [comment]
Please check them one by one and carefully. Please not modify BSP.

**Pull request closed without merging**